oracle/oracle-how-to/deploy-oke-nodes-using-ubuntu-images.rst

```diff
@@ -0,0 +1,250 @@
+
+Creating Managed OKE nodes with Ubuntu
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Managed nodes are node instances whose lifecycle is managed by the OKE service. Since this is a Limited Availability release of Ubuntu images for OKE, you can only create managed nodes through the Oracle Cloud API (``oci`` CLI or SDK). The ability to create managed from nodes from the Oracle Cloud UI will be added at a later.
+
+To create a managed node, start by copying the following cloud-init script into a file called ``user-data.yaml``.
+
+.. code:: yaml
+
+   #cloud-config
+   
+   runcmd:
+     - oke bootstrap
+
+Then, create a placement configuration file to specify where in Oracle Cloud the Managed node pool should be created and save the file as ``placement-config.json``.
+
+.. code:: json 
+
+   [{
+     "compartmentId":"<compartment-id>",
+     "availabilityDomain":"<availability-domain>",
+     "subnetId":"<subnet-id>"
+   }]
+
+
+Lastly, replace the values and run the following command to create the Managed node pool:
+
+.. code:: bash
+   
+  oci ce node-pool create \
+    --cluster-id=<cluster-id> \
+    --compartment-id=<compartment-id> \
+    --name=<pool-name> \
+    --node-shape=<node-shape> \
+    --size=<pool-count> \
+    --kubernetes-version="1.29.1" \
+    --node-image-id=<ubuntu-image-id> \
+    --placement-configs="$(cat placement-config.json)" \
+    --node-metadata='{"user_data": "'"$(base64 user-data.yaml)"'"}'
+
+
+View the node pool status in Oracle Cloud by navigating to :guilabel:`Kubernetes Clusters (OKE)` and click on your cluster, then select :guilabel:`Resources` > :guilabel:`Node pools` and click on the latest node pool.
+
+Everything will be running as expected when the :guilabel:`Kubernetes node condition` and :guilabel:`Node state` of all the nodes are labelled :guilabel:`Ready`.
+
+Creating Self-Managed OKE nodes with Ubuntu
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following instructions assumes that you have configured your OKE cluster to work with self-managed nodes. If you have not done this, refer to the Oracle documentation for `working with self-managed nodes`_
+
+Before adding a self-managed node, ensure you have configured ``kubectl`` to your OKE cluster with the following command. This process will be easier if ``kubectl`` is configured for a single OKE cluster.
+
+.. code:: bash
+  
+  kubectl cluster-info
+
+Next, the self-managed node will need a custom cloud-init script which needs some specific values, namely a Kubernetes certificate from the OKE cluster and the Kubernetes API private endpoint.
+
+Obtain the Kubernetes certificate using ``kubectl`` with the following command and note that ``[0]`` is the index of the cluster if only one is configured:
+
+.. code:: bash
+
+   kubectl config view --raw -o json | jq -r '.clusters[0].cluster.certificate-authority-data'
+
```

Comment:
Maybe include a statement about how to determine your cluster index in cases where multiple clusters are configured.

Reply:
Added `--minify` \[0\] flag so the index isn't needed anymore.

In summary, `kubectl` can have multiple "contexts" which represent different clusters the `kubectl` client can point to/communicate with. Only _one_ context may be active at a time, hence why we needed the index before to select the active option. With `--minify` it trims the data to only return the data for the currently active context.
 
Refs:
- \[0\]: https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_view/

Reply:
I think that's good and less confusing. Thank you!